R/meltAssay.R

```diff
@@ -78,56 +86,60 @@ setGeneric("meltSE",
+                    col.name = sample_name,
                     sample_name = "SampleID",
                     ...)
                standardGeneric("meltSE")
 )
 
-.norm_add_row_data <- function(add_row_data, x, feature_name){
-    if(is.null(add_row_data)){
+.norm_add_row_data <- function(add.row, x, row.name){
+    if(is.null(add.row)){
         return(NULL)
     }
-    if(anyNA(add_row_data)){
-        stop("'add_row_data' contains NA.", call. = FALSE)
+    if(anyNA(add.row)){
+        stop("'add.row' contains NA.", call. = FALSE)
     }
     cn <- colnames(rowData(x))
-    if(is.logical(add_row_data) && length(add_row_data) == 1L && add_row_data){
-        add_row_data <- cn
-    } else if (isFALSE(all(add_row_data %in% cn))) {
-        stop("Please provide valid column names with 'add_row_data' matching ",
+    if(is.logical(add.row) && length(add.row) == 1L && add.row){
+        add.row <- cn
+    } else if (isFALSE(all(add.row %in% cn))) {
+        stop("Please provide valid column names with 'add.row' matching ",
              "those in 'rowData(x)'", call. = FALSE)
     }
-    if(!is.null(rownames(x)) && feature_name %in% add_row_data){
-        warning("'x' contains a column '",feature_name,"' in its ",
+    if(!is.null(rownames(x)) && row.name %in% add.row){
+        warning("'x' contains a column '",row.name,"' in its ",
                 "rowData(), which will ",
-                "be renamed to '",feature_name,"_row'", call. = FALSE)
+                "be renamed to '",row.name,"_row'", call. = FALSE)
     }
-    add_row_data
+    add.row
 }
 
-.norm_add_col_data <- function(add_col_data, x, sample_name){
-    if(is.null(add_col_data)){
+.norm_add_col_data <- function(add.col, x, col.name){
+    if(is.null(add.col)){
         return(NULL)
     }
-    if(anyNA(add_col_data)){
-        stop("'add_col_data' contains NA.", call. = FALSE)
+    if(anyNA(add.col)){
+        stop("'add.col' contains NA.", call. = FALSE)
     }
     cn <- colnames(colData(x))
-    if(is.logical(add_col_data) && length(add_col_data) == 1L && add_col_data){
-        add_col_data <- cn
-    } else if (isFALSE(all(add_col_data %in% cn))) {
-        stop("Please provide valid column names with 'add_col_data' matching ",
+    if(is.logical(add.col) && length(add.col) == 1L && add.col){
+        add.col <- cn
+    } else if (isFALSE(all(add.col %in% cn))) {
+        stop("Please provide valid column names with 'add.col' matching ",
              "those in 'colData(x)'", call. = FALSE)
     }
-    if(!is.null(colnames(x)) && sample_name %in% add_col_data){
-        warning("'x' contains a column '",sample_name,"' in its ",
+    if(!is.null(colnames(x)) && col.name %in% add.col){
+        warning("'x' contains a column '",col.name,"' in its ",
                 "colData(), which will ",
-                "be renamed to '",sample_name,"_col'", call. = FALSE)
+                "be renamed to '",col.name,"_col'", call. = FALSE)
     }
-    add_col_data
+    add.col
```

Comment:
Here also, add return()

Reply:
I think that is good practice. R functions returns always something, I guess it is the value that was used the latest. However, that is not explicit and return() highlights the end of the function

Reply:
ACK,.